spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-pulsar4/build.gradle

```diff
@@ -0,0 +1,27 @@
+plugins {
+	id "java"
+	id "org.springframework.boot.docker-test"
+}
+
+description = "Spring Boot Pulsar smoke test"
+
+configurations.all {
+	resolutionStrategy.eachDependency {
+		if (it.requested.group == 'org.apache.pulsar' &&
+				!(it.requested.name.startsWith('pulsar-client-reactive'))) {
+			it.useVersion '4.0.1'
+		}
+	}
+}
```

Comment:
I really really do not like the fact that this is 99.98% the same as `spring-boot-smoke-test-pulsar`.  These lines are the **ONLY** difference between the modules. I wanted to keep this PR simple but if there is an approach already in place to respect DRY that would be a great polish commit on this.

Reply:
I think in this case the duplication is fine. The samples aren't published and we can drop this one in Boot 3.5 and just revert to having a single smoke test.